backend/src/db/providers/courseProvider.ts

```diff
@@ -8,19 +9,57 @@ import {
   NO_FILTER,
 } from 'config/constants';
 import CourseModel from 'db/models/Course';
+import ClientCourseModel from 'db/models/ClientCourses';
 import { ICourse } from 'interfaces/Ientities/Icourses';
+import { TCourseFields } from 'interfaces/Ientities/IclientCourses';
 import { ICourseStatus, IQueryCourses } from 'interfaces/ICourses/IQueryCourses';
 import BadRequestError from 'classes/errors/clientErrors/BadRequestError';
 import NotFoundError from 'classes/errors/clientErrors/NotFoundError';
 import { SortOrder } from 'enums/common';
 import decodeAndFormatSearchParams from 'utils/decode/decodeSearchParams';
-import ClientCourseModel from 'db/models/ClientCourses';
-import { TCourseFields } from 'interfaces/Ientities/IclientCourses';
+
+const generateCourseStatusLookup = (userId: ObjectId | string) => ({
+  $lookup: {
+    from: 'clientCourses',
+    localField: '_id',
+    foreignField: 'course',
+    pipeline: [
+      {
+        $match: {
+          $expr: {
+            $and: [{ $eq: ['$user', new mongoose.Types.ObjectId(String(userId))] }],
+          },
+        },
+      },
+      {
+        $project: {
+          status: 1,
+          _id: 0,
+        },
+      },
+    ],
+    as: 'status',
+  },
+});
 
 interface ICourseWithStatusDb extends ICourse {
   status: [{ status?: string }];
 }
 
+const normalizeAgregatedCourse = (agregatedCourse: ICourseWithStatusDb): ICourseStatus => {
```

Comment:
why this normalizer in provider?

Reply:
This is too local stuff I think. It will never be used somewhere outside this file. So I decided to make it local

Reply:
good